tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc

```diff
@@ -0,0 +1,306 @@
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <fstream>
+#include <iostream>
+#include "tensorflow/lite/micro/examples/micro_speech/audio_provider.h"
+#include "tensorflow/lite/micro/examples/micro_speech/command_responder.h"
+#include "tensorflow/lite/micro/examples/micro_speech/feature_provider.h"
+#include "tensorflow/lite/micro/examples/micro_speech/micro_features/micro_model_settings.h"
+#include "tensorflow/lite/micro/examples/micro_speech/recognize_commands.h"
+#include "tensorflow/lite/schema/schema_generated.h"
+#include "micro_interpreter.h"
+#include "tensorflow/lite/micro/examples/micro_speech/micro_features/tiny_conv_micro_features_model_data.h"
+#include "lite/version.h"
+#include "micro_mutable_op_resolver.h"
+
+constexpr int kNoOfSamples = 512;
+bool g_is_audio_initialized = false;
+constexpr int kAudioCaptureBufferSize = kAudioSampleFrequency * 0.5;
+int16_t g_audio_capture_buffer[kAudioCaptureBufferSize];
+int16_t g_audio_output_buffer[kMaxAudioSampleSize];
+int32_t g_latest_audio_timestamp = 0;
+
+int16_t audio[513];
+int scores_count[4] = {0};
+
+
+
+char filename[50] = "input.wav";
```

Comment:
Can you break the file loading code into something like examples/micro_speech/ceva_bluevox/audio_provider.cc ?
That should help reduce the amount of duplicated code that's needed here.